cake-autorate.sh

```diff
@@ -1455,38 +1448,40 @@ do
 				# If base rate is sustained, increment sustained base rate timer (and break out of processing loop if enough time passes)
 				if (( enable_sleep_function ))
 				then
-					if [[ ${load_condition[dl]} == *idle* && ${load_condition[ul]} == *idle* ]]
-					then
-						if ((sustained_connection_idle))
-						then
-							((
-								t_sustained_connection_idle_us += (t_start_us-t_last_connection_idle_us),
-								t_last_connection_idle_us=t_start_us
-							))
-						else
-							sustained_connection_idle=1 t_last_connection_idle_us=t_start_us
-						fi
-						if ((t_sustained_connection_idle_us > sustained_idle_sleep_thr_us))
-						then
-							change_state_main "IDLE"	
-
-							log_msg "DEBUG" "Connection idle. Waiting for minimum load."
-
-							if ((min_shaper_rates_enforcement))
+					case ${load_condition[dl]}${load_condition[ul]} in
```

Comment:
Nice!

Reply:
Felt clever coming up with that trick :D